sdk/src/common/random.cc

```diff
@@ -0,0 +1,50 @@
+#include "src/common/random.h"
+#include "src/common/platform/fork.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+// Wraps a thread_local random number generator, but adds a fork handler so that
+// the generator will be correctly seeded after forking.
+//
+// See https://stackoverflow.com/q/51882689/4447365 and
+//     https://github.com/opentracing-contrib/nginx-opentracing/issues/52
+namespace
+{
+class TlsRandomNumberGenerator
+{
+public:
+  TlsRandomNumberGenerator() noexcept
+  {
+    Seed();
+    platform::AtFork(nullptr, nullptr, OnFork);
+  }
+
+  static std::mt19937_64 &engine() noexcept { return engine_; }
+
+private:
+  static thread_local std::mt19937_64 engine_;
```

Comment:
This costs 2.5KB per thread.

Reply:
I'd fine switching to a version like this
https://github.com/lightstep/lightstep-tracer-cpp/blob/master/src/common/fast_random_number_generator.h
It's smaller and faster. And for the use-case of generating unique IDs, I think, perfectly fine.

Reply:
I switched to FastRandomNumberGenerator.

It takes up only 16 bytes and here's the benchmark comparison
https://2145-193156399-gh.circle-artifacts.com/0/benchmark/sdk/test/common/random_benchmark_result.txt
```
2020-03-27 02:18:54
Running bazel-out/host/bin/sdk/test/common/random_benchmark
Run on (36 X 3370.14 MHz CPU s)
CPU Caches:
  L1 Data 32K (x18)
  L1 Instruction 32K (x18)
  L2 Unified 1024K (x18)
  L3 Unified 25344K (x1)
Load Average: 6.47, 9.48, 13.46
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_RandomIdGeneration          2.13 ns         2.13 ns     63421653
BM_RandomIdStdGeneration       6.18 ns         6.18 ns     22676378
```
The algorithm [xorshift](https://en.wikipedia.org/wiki/Xorshift) is fine for ID generation.

Reply:
Nice. Thanks for adding a benchmark also.